com.unity.render-pipelines.universal/Shaders/Utils/StencilDeferred.shader

```diff
@@ -184,8 +184,12 @@ Shader "Hidden/Universal Render Pipeline/StencilDeferred"
                 unityLight.shadowAttenuation = 1.0;
             else
             {
-                #if defined(_MAIN_LIGHT_SHADOWS)
-                    float4 shadowCoord = TransformWorldToShadowCoord(posWS.xyz);
+                #if defined(MAIN_LIGHT_CALCULATE_SHADOWS)
+                    #if defined(_MAIN_LIGHT_SHADOWS_SCREEN)
+                        float4 shadowCoord = float4(screen_uv, 0.0, 1.0);
```

Comment:
@kaychang-unity In case of 'Screen Space Shadow', it needs screen uv to compute shadowing. I tried to use 'ComputeScreenPos(input.positionCS.xy), but it doesn't work correctly and also tried 'ComputeScreenPos(TransformWorldToHClip(posWS)), it works well.

but the second needs more computation and I think this is not good idea. So I put directly 'screen_us' for the performance.

I can't understand why that positionCS was different between forward pass and deferred pass

Reply:
I think using `screen_uv` should be the fastest way to sample screen space texture. It's already done for other gbuffer textures, so you should keep using `screen_uv` instead.

As for `ComputeScreenPos(input.positionCS.xy)`, I see that is it called only from vertex shader, not from fragment shader. In vertex shader, `input.positionCS.xy` will store vertex position in homogeneous space but before perspective-divide by `input.positionCS.w`. So the value after perspective-divide by `input.positionCS.w` will be in range [-1; 1]. In pixel shader, `input.positionCS.xy` stores actual pixel coordinates (not the same as vertex shader), which will be different. I presume that is why `ComputeScreenPos(input.positionCS.xy)` does not work for you in the pixel shader.

Reply:
Ahh.. in fragment shader positionCS is normalized device coordinate! Thanks.

Reply:
In fragment, shader positionCS.xy is pixel coordinates. For instance, if the render-target resolution is 1280x720, it will be x=[0;1280], y=[0;720] (with half pixel offset).
positionCS is normalized device coordinate in vertex shader only.